NachoSpigot-Server/src/main/java/net/minecraft/server/EntityLiving.java

```diff
@@ -1789,7 +1790,28 @@ public void receive(Entity entity, int i) {
     }
 
     public boolean hasLineOfSight(Entity entity) {
-        return this.world.rayTrace(new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ), new Vec3D(entity.locX, entity.locY + (double) entity.getHeadHeight(), entity.locZ)) == null;
+        if (entity instanceof EntityPlayer && NachoConfig.enableImprovedHitReg) {
+            // Head height is 1,5725
+            // Split it into three to get a more accurate line of sight -> 0.52416667
+
+            double parts = entity.getHeadHeight() / 3;
+
+            return this.world.rayTrace(
+                    new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
+                    new Vec3D(entity.locX, entity.locY + (parts * 3), entity.locZ)
+            ) == null || this.world.rayTrace(
+                    new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ),
```

Comment:
avoid creating unneed objects here  new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ) can be reused